mm/mm_heap/mm_extend.c

```diff
@@ -77,7 +77,7 @@ void mm_extend(FAR struct mm_heap_s *heap, FAR void *mem, size_t size,
 
   /* Take the memory manager semaphore */
 
-  mm_takesemaphore(heap);
+  DEBUGVERIFY(mm_takesemaphore(heap));
```

Comment:
The statement should not be wrapped in the DEBUGVERIFY macro, otherwise it will be discarded when `CONFIG_DEBUG_ASSERTIONS` is disabled.